channel.go

```diff
@@ -57,7 +57,9 @@ type ChannelOptions struct {
 	// The name of the process, for logging and reporting to peers
 	ProcessName string
 
-	// OnPeerStatusChanged
+	// OnPeerStatusChanged is an optional callback that receives a notification
```

Comment:
I’m hacking out the YARPC connection manager for TChannel and so far it looks like receiving a notification when the last connection dies is necessary and sufficient. Since I’m calling peer.Connect(), I have a direct line on when the connection count goes form 0 to 1.

The decrement case would be useful if we sought to ensure a pool of redundant connections, which I might in a later pass.